src/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs

```diff
@@ -136,7 +136,16 @@ private void Initialize()
                 // machine connecting to public server).
 
                 // SMTP RFC's require ASCII only host names in the HELO/EHLO message.
-                string clientDomainRaw = IPGlobalProperties.GetIPGlobalProperties().HostName;
+                string clientDomainRaw;
+                try
+                {
+                    clientDomainRaw = IPGlobalProperties.GetIPGlobalProperties().HostName;
+                }
+                catch (NotSupportedException)
```

Comment:
Please add a comment with a TODO here.  We plan to support this during our work on UWP for .NET Core 2.1 milestone.  Once System.Net.NetworkInformation is working on UWP, this won't be returning NotSupportedException anymore.